test/Microsoft.AspNetCore.Server.KestrelTests/MemoryPoolIteratorTests.cs

```diff
@@ -491,7 +584,36 @@ public void GetsKnownVersion(string input, char endChar, bool expectedResult, st
             Assert.Equal(expectedResult, result);
             Assert.Equal(expectedKnownString, knownString);
 
+            // Test at boundary
+            var maxSplit = Math.Min(input.Length, 9);
+            var nextBlock = _pool.Lease();
+
+            for (var split = 0; split <= maxSplit; split++)
```

Comment:
I considered making an AtBlockBoundary() variation of this test but then there would be 8 times as many InlineData attributes as this test. I decided it wasn't worth it.